sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2349,7 +2350,22 @@ class Analyzer(
   }
 
   /**
-   * Replace [[TimeZoneAwareExpression]] without timezone id by its copy with session local
+   * Replace unresolved expressions in grouping keys with resolved ones in SELECT clauses.
+   */
+  object ResolveAggAliasInGroupBy extends Rule[LogicalPlan] {
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+      case agg @ Aggregate(groups, aggs, child)
+          if conf.groupByAliases && child.resolved && groups.exists(!_.resolved) =>
+        agg.copy(groupingExpressions = groups.map {
+          case u: UnresolvedAttribute => aggs.find(ne => resolver(ne.name, u.name)).getOrElse(u)
```

Comment:
I checked a behaviour in PostgreSQL and I found an error below happened in that case;
```
postgres=# select count(value) as k from t group by k;
ERROR:  aggregate functions are not allowed in GROUP BY at character 8     
``` 
So, I fixed code to throw an exception ASAP in that case. How about this fix? @viirya